File: Cpp/643_Maximum_Average_Subarray_I.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        int max_sum = 0;
        int sum = 0;
        for(int i=0; i<k; ++i){
            sum += nums[i];
        }
        max_sum = sum;
        for(int i=k; i<nums.size(); ++i){
            sum += nums[i] - nums[i-k];
            max_sum = max(max_sum, sum);
        }
        
        return static_cast<double>(max_sum) / k;
    }
};

File: Kotlin/643_Maximum_Average_Subarray_I.kt
class Solution {
    fun findMaxAverage(nums: IntArray, k: Int): Double {
        var sum = 0
        for(i in 0 until k){
            sum += nums[i]
        }
        
        var maxSum = sum
        
        for(i in k until nums.size){
            sum += nums[i]
            sum -= nums[i - k]
            maxSum = maxOf(sum, maxSum)
        }
        
        return maxSum / k.toDouble()
    }
}